crates/bevy_scene/src/dynamic_scene_builder.rs

```diff
@@ -234,4 +244,22 @@ mod tests {
         scene_entities.sort();
         assert_eq!(scene_entities, [entity_a_b.id(), entity_a.id()]);
     }
+
+    #[test]
+    fn remove_componentless_entity() {
+        let mut world = World::default();
+
+        let atr = AppTypeRegistry::default();
+        atr.write().register::<ComponentA>();
+        world.insert_resource(atr);
+
+        let entity = world.spawn(ComponentB).id();
+
+        let mut builder = DynamicSceneBuilder::from_world(&world);
+        builder.extract_entity(entity);
+        builder.remove_empty_entities();
+        let scene = builder.build();
+
+        assert_eq!(scene.entities.len(), 0);
```

Comment:
It might be good to also create an entity that has components and then assert the entity with components doesn't get removed here